### PR TITLE
idea: codespell the codebase

### PR DESCRIPTION
**Pull request by yarikoptic**
We keep running into typos, and apparently even within interface. So what if we prevent typos altogether?

I am surprised in how well codespell performed!  There seems to be no inline ignores supported yet, but specific lines can be excluded: https://github.com/codespell-project/codespell/issues/1212

if we decide to proceed with this:
- [x] review changes for no false positives
- ~~add deprecation cycle for commitish -> committish fix~~ decided to not bother changing
- [x] add github action which would fail if diff introduces a typo
- [x] rm TEMP commit which was to test workflow to fail

WDYT?

### CODE REVIEW

**Review by kyleam — COMMENTED** [review]
> review changes for no false positives

Glancing through, most of these look good to me.  I'd vote to just leave "commitish" alone.  It's true that Git uses only "commit-ish" and "commit**t**ish" [*], but I think using a slightly different variant of a made-up word is fine and not worth deprecation cycles.  But, if we do go with the double-t form, it looks like there are single-t variants (some plural) left after this PR's changes.

[*] The single-t variant was introduced before the double-t in this code base, so I'd guess internal consistency was the reason the single-t variant ended up more frequent despite git's preference.

File: datalad/support/tests/test_network.py [hunk 1]
```diff
@@ -333,8 +333,8 @@ def test_url_samples():
         weired_url = RI(weired_str)
         repr(weired_url)
         cml.assert_logged(
-            'Parsed version of SSHRI .weired:/. '
-            'differs from original .weired://.'
+            'Parsed version of SSHRI .weird:/. '
```
[the remark this answers is not in the document]
Comment (yarikoptic) [reply datalad/support/tests/test_network.py:336]: thanks - I will fix for that if we decide to proceed

**Comment by yarikoptic** [comment]
ha ha -- indeed we have plural forms as well and even in the function signatures:
```shell
$> git grep 'def .*commitish'
datalad/support/gitrepo.py:    def format_commit(self, fmt, commitish=None):
datalad/support/gitrepo.py:    def get_hexsha(self, commitish=None, short=False):
datalad/support/gitrepo.py:    def commit_exists(self, commitish):
datalad/support/gitrepo.py:    def get_merge_base(self, commitishes):
datalad/support/gitrepo.py:    def describe(self, commitish=None, **kwargs):
```
I would have preferred to harmonize and make it into double-t form, but  indeed the game is not worth the candles here IMHO too.  So, ok - we should be able to exclude it via `--ignore-regex commitish`.

But what about overall idea to automate/mandate absent errors? 
I see the pain point of possible need to extend that file with exact line matches and also since we are not using pre-commit and unlikely anyone would run the check locally, to catch those only at the PR stage. Just do not want to impose unless there is general agreement.

FWIW - we added codecheck to pre-commit and linting check within dandi-cli today, will see how it goes.

**Comment by kyleam** [comment]
> But what about overall idea to automate/mandate absent errors?

In my opinion there not much harm in trying it for a while and seeing how it goes.  Its initial pass looks pretty good.

**Comment by mih** [comment]
I am with @kyleam -- let's try

**Comment by yarikoptic** [comment]
Great, thanks @kyleam and @mih. I will do it then. I think even would be worth against maint since we decided to keep commitish alone and iirc no other notable api changes

**Comment by yarikoptic** [comment]
if we are to aim for the `master` for this -- it is ready.

**Comment by kyleam** [comment]
> edit: decided not to bother with `maint` since lots of removals happened, so not worth then chasing conflicts imho.

Makes sense.  Thanks @yarikoptic

**Pull request merged**